flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/AbstractOperatorRestoreTestBase.java

```diff
@@ -104,16 +123,32 @@ public abstract class AbstractOperatorRestoreTestBase extends TestLogger {
                             .setNumberSlotsPerTaskManager(NUM_SLOTS_PER_TM)
                             .build());
 
-    private final boolean allowNonRestoredState;
     private final ScheduledExecutor scheduledExecutor =
             new ScheduledExecutorServiceAdapter(EXECUTOR_RESOURCE.getExecutor());
 
-    protected AbstractOperatorRestoreTestBase() {
-        this(true);
+    protected AbstractOperatorRestoreTestBase(FlinkVersion flinkVersion) {
+        this.flinkVersion = flinkVersion;
     }
 
-    protected AbstractOperatorRestoreTestBase(boolean allowNonRestoredState) {
-        this.allowNonRestoredState = allowNonRestoredState;
+    protected void internalGenerateSnapshots(FlinkVersion targetVersion) throws Exception {
```

Comment:
I'm a bit puzzled here: Where is the data generation been done in the old code? :thinking:

Reply:
Ok, after I have done the data generation myself in FLINK-31593, I get how it should be implemented :+1: